base/strings/basic.jl

```diff
@@ -802,7 +802,7 @@ elsize(s::Type{<:CodeUnits{T}}) where {T} = sizeof(T)
 @propagate_inbounds getindex(s::CodeUnits, i::Int) = codeunit(s.s, i)
 IndexStyle(::Type{<:CodeUnits}) = IndexLinear()
 @inline iterate(s::CodeUnits, i=1) = (i % UInt) - 1 < length(s) ? (@inbounds s[i], i + 1) : nothing
-
+strides(::CodeUnits) = (1,)
```

Comment:
```suggestion
strides(::CodeUnits{<:Any, <:Union{String, SubString{String}}}) = (1,)
```

Reply:
I actually think this is true for all codeunits. You can't have a codeunits with a step which is not one.
Or does strides specifically refer to memory, not the step in general?

Reply:
> Or does strides specifically refer to memory, not the step in general?

Yes, that's what strides means.  **All** arrays must have sequential indexing.

Reply:
It might be possible that similarly adding `pointer` would recover some of the test failures, but I've not looked into them yet.